sweep/aggregator.go

```diff
@@ -21,12 +21,106 @@ const (
 	DefaultFeeRateBucketSize = 10
 )
 
+// inputCluster is a helper struct to gather a set of pending inputs that
+// should be swept with the specified fee rate.
+type inputCluster struct {
+	lockTime     *uint32
+	sweepFeeRate chainfee.SatPerKWeight
+	inputs       pendingInputs
+}
+
+// createInputSets goes through the cluster's inputs and constructs sets of
+// inputs that can be used to generate a sweeping transaction. Each set
+// contains up to the configured maximum number of inputs. Negative yield
+// inputs are skipped.  No input sets with a total value after fees below the
+// dust limit are returned.
+func (c *inputCluster) createInputSets(maxFeeRate chainfee.SatPerKWeight,
+	maxInputs int) []InputSet {
+
+	// Turn the inputs into a slice so we can sort them.
+	inputList := make([]*pendingInput, 0, len(c.inputs))
+	for _, input := range c.inputs {
+		inputList = append(inputList, input)
+	}
```

Comment:
feels like we should have a `Values()` method for this.

Reply:
yeah would worry too much about the old cluster tho.